crates/fluvio-stream-model/src/store/dual_store.rs

```diff
@@ -403,6 +408,21 @@ mod listener {
 
     use super::{LocalStore, Spec, MetadataItem, MetadataChanges};
 
+    pub struct AtLeastOneChangeListener<'a> {
+        event_publisher: &'a EventPublisher,
+    }
+
+    impl<'a> AtLeastOneChangeListener<'a> {
+        pub fn new(event_publisher: &'a EventPublisher) -> Self {
+            Self { event_publisher }
+        }
+        pub async fn listen_for_at_least_one_change(&'a self) {
```

Comment:
is there reason why this is async since there is no await here.

Reply:
> Most of changes make sense except concern about `AtLeastOneChangeListener`. I do think it needs unit test

Agreed. Definitely needs a unit test.

Reply:
> is there reason why this is async since there is no await here.

Yes. async is just syntatic sugar for a fn that returns a future. In this case it returns an `EventListener` https://docs.rs/event-listener/2.5.3/event_listener/struct.EventListener.html, which impls `IntoFuture`

We don't want to await here as it would block the thread and disable the timeout, so we return the future.

Reply:
Shouldn't  `listen` fn should be like this?
```
pub async fn listen(&'a self) {
      while self.event_publisher.current_change() <= 0 {
              self.event_publisher.listen().await;
      }
 }
```

Otherwise, I don't see how executor can poll since there is no return in while loop.  Maybe unit test can prove that.

Reply:
Yes, you are right. It loops, not returns.

Reply:
Needs to be fixed.

Reply:
Fixed + unit test

Reply:
At this point, AtLeastOneChangeListener is just a copy paste of ChangeListener. I think it was better with just one. We can still keep the unit tests.

Reply:
Let's do that.